File: easy/football.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my $i = 1;
    my @r;
    my @u;
    my %t;
    map {
        if ( $_ eq '|' ) { $i++; }
        else {
            if ( exists $t{$_} ) { $t{$_} .= ',' . $i; }
            else {
                $t{$_} = $i;
                push @u, $_;
            }
        }
    } split( / /, $line );
    @u = sort { $a <=> $b } @u;
    for my $j ( 0 .. $#u ) {
        push @r, sprintf( "%d:%s;", $u[$j], $t{ $u[$j] } );
    }
    printf( "%s\n", join( ' ', @r ) );
}
close(INFILE);

File: easy/football.rb
Team = Struct.new(:id, :countries)

File.open(ARGV[0]).each_line do |line|
  s = line.split(' | ').map(&:split)
  t = []
  r = []
  s.each_with_index do |i, ix|
    i.each do |j|
      jx = j.to_i
      f = false
      (0...t.length).each do |k|
        next if t[k].id != jx
        t[k].countries += format(',%d', ix + 1)
        f = true
        break
      end
      t << Team.new(jx, (ix + 1).to_s) unless f
    end
  end
  t.sort { |x, y| x.id <=> y.id }.each do |i|
    r << format('%d:%s;', i.id, i.countries)
  end
  puts r.join(' ')
end
